src/ansys/simai/core/utils/requests.py

```diff
@@ -76,6 +77,21 @@ def handle_http_errors(response: requests.Response) -> None:
             raise ApiClientError(f"{response.status_code}: {response.reason}", response) from e
 
 
+@overload
+def handle_response(response: requests.Response, return_json: Literal[True]) -> JSON:
+    ...
+
+
+@overload
+def handle_response(response: requests.Response, return_json: Literal[False]) -> requests.Response:
+    ...
+
+
+@overload
+def handle_response(response: requests.Response, return_json: bool) -> APIResponse:
+    ...
```

Comment:
```suggestion
```

You don't need this overload since it's exactly the same as the function signature. Also all the cases are covered by the other overloads.

Reply:
I know it looks dumb but the documentation says you do.